Utils/src/HiddenSectorProducer.cc

```diff
@@ -367,13 +626,237 @@ void HiddenSectorProducer::produce(edm::StreamID, edm::Event& iEvent, const edm:
+    
+    // make list of "dark hadron subjets" in reco particles using the match to gen
+    int unmatchedGen = 0;
+    std::vector<std::vector<CLorentzVector> > recoSubjets;
+    std::vector<CLorentzVector> recoDarkHadronJets;
+    std::vector<int> flatIndex;
+    std::vector<int> flatGenPdgId;
+    std::vector<int> flatRecoPdgId;
+    std::vector<int> RecoMatchStage;
+    std::vector<std::vector<int>> matchStage;
+    std::vector<std::vector<int>> RecoPdgid;
+
+    for(unsigned i = 0; i < matchedRecoIndex.size(); i++){
+      std::vector<CLorentzVector> recoSubjetConstituents;
+      CLorentzVector tmpJet;
+      std::vector<int> tmpMatchStage;
+      std::vector<int> tmpRecoPdgid;
+      for(unsigned j = 0; j < matchedRecoIndex[i].size(); j++){
+	if (matchedRecoIndex[i][j] != -1) {
+	  flatIndex.push_back(matchedRecoIndex[i][j]);
+	  flatGenPdgId.push_back(GenJets_darkHadronJets_constituentsPdgid->at(genJetIndex)[i][j]);
+	  flatRecoPdgId.push_back(cands_pdgids[recoJetIndex][matchedRecoIndex[i][j]]);
+	  RecoMatchStage.push_back(matchStageReco[matchedRecoIndex[i][j]]);
+	  tmpMatchStage.push_back(matchStageReco[matchedRecoIndex[i][j]]);
+	  tmpRecoPdgid.push_back(cands_pdgids[recoJetIndex][matchedRecoIndex[i][j]]);
+	  tmpJet += jets_cands[recoJetIndex][matchedRecoIndex[i][j]];
+	  recoSubjetConstituents.emplace_back(jets_cands[recoJetIndex][matchedRecoIndex[i][j]]);
+	}
+	else {
+	  unmatchedGen++;
+	}
+      }
+      recoDarkHadronJets.emplace_back(tmpJet);
+      recoSubjets.emplace_back(recoSubjetConstituents);
+      matchStage.emplace_back(tmpMatchStage);
+      RecoPdgid.emplace_back(tmpRecoPdgid);
+    }
+
+    GenJets_constituents_pdgid->at(genJetIndex) = flatGenPdgId;
+    GenJets_constituents_matchIndex->at(genJetIndex) = flatIndex;
+    GenJets_nConstituents_unmatched->at(genJetIndex) = unmatchedGen;
+    Jets_darkHadronJets->push_back(recoDarkHadronJets);
+    Jets_darkHadronJets_constituents_genMatchOnly->push_back(recoSubjets);
+    Jets_darkHadronJets_constituentsMatchStage->push_back(matchStage);
+    Jets_darkHadronJets_constituentsPdgid->push_back(RecoPdgid);
+    Jets_constituents_pdgid->push_back(flatRecoPdgId);
+    Jets_constituents_matchStageReco->push_back(RecoMatchStage);
+
+    //For unmatched reco particles, assign them to the closest dark hadron, and the second closest to demonstrate an uncertainty in this procedure
+    std::vector<std::vector<CLorentzVector> > recoSubjetsNextDH = recoSubjets;
+    std::vector<std::vector<CLorentzVector> > newRecoSubjets = recoSubjets;
+    std::vector<int> recoUnmatchedPdgid = {};
+    std::vector<int> isAssigned;
+    int nRecoUnmatched = 0;
+    for(unsigned i = 0; i < recoMatched.size(); i++){
+      isAssigned.push_back(0);
```

Comment:
could just do `std::vector<int> isAssigned(recoMatched.size(), 0)`? (before the loop)